gitcoinco/grants-stack
# Finalize Round V1 / V2

fixes https://github.com/gitcoinco/grants-stack/issues/2981
fixes https://github.com/gitcoinco/grants-stack/issues/2978
fixes https://github.com/gitcoinco/grants-stack/issues/2979

## Description

This one is a bit difficult to QA because of the many steps involved in creating a finalizable round. I have done those steps and recorded a Loom finalizing a round: https://www.loom.com/share/6aaadd27f07846c8bbf1c2e12b6c3592

I tested in local because I had to hack a few things to get some proper matching results with the amount of votes I had.

At the end of the video I noticed that the Fund Grantees page was not working, I found the issue and fixed it:
<img width="1182" alt="Screenshot 2024-03-13 at 13 28 46" src="https://github.com/gitcoinco/grants-stack/assets/711886/c7d45e9f-9410-4cda-83b6-4efd9d20fb9f">

I also paid out a grantee to make sure the uploaded Merkle root is valid:

<img width="1206" alt="Screenshot 2024-03-13 at 13 46 30" src="https://github.com/gitcoinco/grants-stack/assets/711886/c97072e8-01f4-4879-9c07-93b0675ae831">

I suggest that we leave the QA for Allo v2 when we do the internal E2E round next week, just because the donation flow for Allo v2 is not done yet.

## How to test

- Create a finalizable round (create round, apply, approve, vote, fund, wait for round end)
- Click on Finalize matches
- Verify that payout screen is available
- Verify that finalized matches show




## Checklist

This PR:

- [x] Does it add new payout or donation token? In this case, have they been added to the indexer to avoid it to crash?
